arbitrum-docs/for-devs/quickstart-solidity-hardhat.md

```diff
@@ -0,0 +1,491 @@
+| **Presentation** (UI) | Served from **Offchain's servers**. Rendered and executed by your **browser**.          | <- same                                                                                    | <- same                                                                                | <- same                                                                                   |
+| **Money**             | Devs and users pay centralized service providers for server access using fiat currency. | <- same, but only for the presentation-layer concerns (code that supports frontend UI/UX). | <- same, but devs and users pay **testnet $ETH** to testnet validators.                | <- same, but instead of testnet $ETH, they use **mainnet $ETH**.                          |
+
+
+Next, we'll deploy our smart contract to a network of real nodes: Arbitrum's Goerli testnet.
+
+### Deploy the smart contract to the Arbitrum Goerli testnet
+
+We were able to deploy to a local testnet for free because we were using [Hardhat's built-in Ethereum network emulator](https://hardhat.org/hardhat-network/docs/overview#hardhat-network). Because Arbitrum's Goerli testnet is powered by a real network of real nodes, we'll need to pay a small transaction fee to deploy our smart contract. This fee will be paid with the Arbitrum Goerli testnet's token, $AGOR.
+
+:::info $AGOR ISN'T REAL
+
+$AGOR isn't a canonical term, or a real token. It's just convenient shorthand for "Arbitrum Goerli testnet ETH" that we use to make this quickstart more readable.
+
+:::
+
+First, update the `hardhat.config.js` file to specify the private key of the test account that you'll use to deploy your smart contract (and pay the transaction fee):
+
+```javascript title="hardhat.config.js"
+// ...
+const GOERLI_TESTNET_PRIVATE_KEY = ""; // <- this should **not** begin with "0x"
+// ...
+accounts: [GOERLI_TESTNET_PRIVATE_KEY] // <- uncomment this line
+// ...
+```
+
+:::caution
+
+Note that we're adding a private key to a config file. This is **not** a best practice. In a real application, you should use a [signer](https://docs.ethers.io/v5/api/signer/) to sign transactions instead of handling private keys directly.
+
+:::
+
+Next, let's deposit some $AGOR into the wallet corresponding to the private key we added to `hardhat.config.js`. There are a few ways to acquire $AGOR:
+
+1. Use an L2 Goerli ETH faucet.
+2. Use an L1 Goerli ETH faucet and bridge your L1 Goerli ETH into Arbitrum L2 using [the Arbitrum bridge](https://bridge.arbitrum.io/).
+   - [goerlifaucet.com](https://goerlifaucet.com/) is a popular L1 Goerli ETH faucet.
+3. Ask someone nicely on [Discord](https://discord.gg/ZpZuw7p)[^6].
```

Comment:
I'm wondering if we want to push people to ask us for faucets on Discord or not :?

Reply:
Agree. Also, if there isn't any L2 Goerli faucet I wouldn't mention it here. It might also lead users to ask about it on Discord.